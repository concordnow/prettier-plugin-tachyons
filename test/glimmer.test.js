const fs = require("fs");
const path = require("path");
const prettier = require("prettier");
const plugin = require("..");

const fixture = fs.readFileSync(
  path.join(__dirname, "./__fixtures__/glimmer.fix"),
  "utf-8"
);

function format(content, opts = {}) {
  return prettier.format(content, {
    ...opts,
    parser: "glimmer",
    plugins: [plugin],
  });
}

describe("Test glimmer parser", () => {
  test("basic glimmer component", async () => {
    const formatted = format(fixture);
    expect(formatted).toMatchSnapshot();
  });
});
