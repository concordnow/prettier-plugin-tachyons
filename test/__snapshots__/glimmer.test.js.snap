// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test glimmer parser basic glimmer component 1`] = `
"<article class=\\"\\">
  <h1 class=\\"\\">Glimmer test</h1>
  <p class=\\"\\">You have clicked the button {{this.count}} times.</p>
  <button class=\\"\\" type=\\"button\\" {{on \\"click\\" this.increment}}>
    Click
  </button>
</article>"
`;
